toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -967,3 +975,175 @@ func checkEnvironmentVars() error {
 
 	return nil
 }
+
+func customizeUkiImageHelper(buildDir string, buildImageFile string) error {
+	logger.Log.Debugf("Customizing UKI")
+
+	var err error
+
+	loopback, err := safeloopback.NewLoopback(buildImageFile)
+	if err != nil {
+		return fmt.Errorf("failed to connect to image file to provision UKI:\n%w", err)
+	}
+	defer loopback.Close()
+
+	diskPartitions, err := diskutils.GetDiskPartitions(loopback.DevicePath())
+	if err != nil {
+		return err
+	}
+
+	systemBootPartition, err := findSystemBootPartition(diskPartitions)
+	if err != nil {
+		return err
+	}
+	bootPartition, err := findBootPartitionFromEsp(systemBootPartition, diskPartitions, buildDir)
+	if err != nil {
+		return err
+	}
+
+	systemBootPartitionTmpDir := filepath.Join(buildDir, tmpEspParitionDirName)
+	// Temporarily mount the esp partition.
+	systemBootPartitionMount, err := safemount.NewMount(systemBootPartition.Path, systemBootPartitionTmpDir, systemBootPartition.FileSystemType, 0, "", true)
+	if err != nil {
+		return fmt.Errorf("failed to mount esp partition (%s):\n%w", bootPartition.Path, err)
+	}
+	defer systemBootPartitionMount.Close()
+
+	bootPartitionTmpDir := filepath.Join(buildDir, tmpParitionDirName)
+	// Temporarily mount the boot partition.
+	bootPartitionMount, err := safemount.NewMount(bootPartition.Path, bootPartitionTmpDir, bootPartition.FileSystemType, 0, "", true)
+	if err != nil {
+		return fmt.Errorf("failed to mount partition (%s):\n%w", bootPartition.Path, err)
+	}
+	defer bootPartitionMount.Close()
+
+	// Read the stub path.
+	stubPath := filepath.Join(bootPartitionTmpDir, "linuxx64.efi.stub")
+
+	// Prepare kernel cmdline arguments.
+	grubCfgFullPath := filepath.Join(bootPartitionTmpDir, "grub2/grub.cfg")
+	if err != nil {
+		return fmt.Errorf("failed to stat file (%s):\n%w", grubCfgFullPath, err)
+	}
+
+	// Read the grub.cfg file.
+	grubCfgContent, err := file.Read(grubCfgFullPath)
+	if err != nil {
+		return fmt.Errorf("failed to read file (%s):\n%w", grubCfgFullPath, err)
+	}
+
+	// Split the file content into lines and search for the 'linux' line
+	lines := strings.Split(string(grubCfgContent), "\n")
+	var linuxLine string
+	for _, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmedLine, "linux") {
+			linuxLine = trimmedLine
```

Comment:
I believe that the first two entries here will be "linux" and the kernel's filename. We don't want to pass those as cmdline arguments to Linux, so they should probably be stripped